lib/net/pool.js

```diff
@@ -2012,6 +2104,145 @@ class Pool extends EventEmitter {
+    const filterHashes = packet.filterHashes;
+    let blockHeight = await this.chain.getHeight(stopHash)
+      - filterHashes.length + 1;
+    const stopHeight = await this.chain.getHeight(stopHash);
+    for (const filterHash of filterHashes) {
+      if (blockHeight > stopHeight) {
+        peer.increaseBan(10);
+        return;
+      }
+      const basicFilter = new BasicFilter();
+      basicFilter._hash = filterHash;
+      const filterHeader = basicFilter.header(previousFilterHeader);
+      const lastFilterHeader = this.cfHeaderChain.tail;
+      const blockHash = await this.chain.getHash(blockHeight);
+      const cfHeaderEntry = new CFHeaderEntry(blockHash,
+        filterHeader, lastFilterHeader.height + 1);
+      this.cfHeaderChain.push(cfHeaderEntry);
+      previousFilterHeader = filterHeader;
+      blockHeight++;
+    }
+    this.logger.info('CFHeader height: %d', this.cfHeaderChain.tail.height);
+    if (this.headerChain.tail.height <= stopHeight)
+      this.emit('cfheaders');
+    else {
+      const nextStopHeight = stopHeight + 2000 < this.chain.height
+        ? stopHeight + 2000 : this.chain.height;
+      const nextStopHash = await this.chain.getHash(nextStopHeight);
+      this.requestedStopHash = nextStopHash;
+      peer.sendGetCFHeaders(filterType, stopHeight + 1, nextStopHash);
+    }
+  }
+
+  async handleCFilters(peer, packet) {
+    const filterType = packet.filterType;
+    const indexer = this.getFilterIndexer(filtersByVal[filterType]);
+    if (indexer.height % 100 === 0)
+      this.logger.debug(
+        'Received CFilter 100 packets from %s', peer.hostname()
+      );
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const blockHash = packet.blockHash;
+    this.cfHeaderChain.shift();
+    const filter = packet.filterBytes;
+
+    if (filterType !== this.requestedFilterType) {
+      this.logger.warning('Received CFilter packet with wrong filterType');
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const blockHeight = await this.chain.getHeight(blockHash);
+    const stopHeight = await this.chain.getHeight(this.requestedStopHash);
+
+    if (!(blockHeight >= this.getcfiltersStartHeight
+       && blockHeight <= stopHeight)) {
+      this.logger.warning('Received CFilter packet with wrong blockHeight');
+      peer.increaseBan(10);
+      return;
+    }
+
+    const basicFilter = new BasicFilter();
+    const gcsFilter = basicFilter.fromNBytes(filter);
+
+    const filterHeader = this.cfHeaderChain.head.header;
+    await indexer.saveFilter(blockHash, blockHeight, gcsFilter, filterHeader);
+    const cFilterHeight = await indexer.height;
+    if (cFilterHeight % 100 === 0)
+      this.logger.info('CFilter height: %d', cFilterHeight);
+    this.emit('cfilter', blockHash, gcsFilter);
+    const startHeight = stopHeight + 1;
+    let nextStopHeight;
+    if (cFilterHeight === stopHeight
+      && stopHeight < this.chain.height) {
```

Comment:
```suggestion
    if (cFilterHeight === stopHeight && stopHeight < this.chain.height) {
```

Reply:
I did it in one line only Anmol. But we gotta keep linting in mind as well.

Reply:
I don't think it'll throw a lint error

Reply:
I think the original is best for readability and consistency with codebase although we usually double indent the second condition like so:

https://github.com/bcoin-org/bcoin/blob/6ae1752e5a408e6e35b67841c4fd9b972d9ad613/lib/blockchain/chain.js#L585-L586